src/Cart.php

```diff
@@ -831,7 +831,12 @@ private function isMulti($item)
      */
     private function storedCartWithIdentifierExists($identifier)
```

Comment:
Maybe we should rename this method since it then would also check for the instance.

Reply:
```php
private function storedCartWithIdentifierExists($identifier)
{
    return $this->getConnection()->table($this->getTableName())->where(['identifier' => $identifier, 'instance'=> $this->currentInstance()])->exists();
}
```
Using the current instance directly breaks the `merge` method too.

```php
$cart = $this->getCartDiscount(50);
$cart->add(new BuyableProduct(1, 'Item', 10.00), 1);
$cart->add(new BuyableProduct(2, 'Item 2', 10.00), 1);
$cart->store('test');

$cart2 = $this->getCart();
$cart2->instance('test2');
$cart2->setGlobalTax(0);
$cart2->setGlobalDiscount(0);

$this->assertEquals('0', $cart2->countInstances());
$cart2->merge('test');
$this->assertEquals('2', $cart2->countInstances()); // count is still 0
```
The items in `$cart` are not merged into `$cart2` because the `merge` method checks if the cart to be merged is also of the same instance.

```php
public function merge($identifier, $keepDiscount = false, $keepTax = false, $dispatchAdd = true)
{
    if (!$this->storedCartWithIdentifierExists($identifier)) { //should pass an instance name to this method imo
        return false;
    }

    $stored = $this->getConnection()->table($this->getTableName())
        ->where(['identifier'=> $identifier,'instance'=>$this->currentInstance()])->first(); //i added the instance to it

    $storedContent = unserialize($stored->content);

     foreach ($storedContent as $cartItem) {
         $this->addCartItem($cartItem, $keepDiscount, $keepTax, $dispatchAdd);
     }

    $this->events->dispatch('cart.merged');

     return true;
}
```

Fixing this means we'll have to change the signature of the `storedCartWithIdentifierExists` method to something like `storedCartWithIdentifierExists($instance,$identifier)` this way we can pass the instance of the cart to be merged too.

I think we should pass the instance as the last parameter with a default value to avoid a breaking change.